presto-main/src/main/java/com/facebook/presto/sql/planner/PlanOptimizers.java

```diff
@@ -269,6 +272,9 @@ public PlanOptimizers(Metadata metadata, SqlParser sqlParser, FeaturesConfig fea
                         ImmutableList.of(new com.facebook.presto.sql.planner.optimizations.EliminateCrossJoins()), // This can pull up Filter and Project nodes from between Joins, so we need to push them down again
                         ImmutableSet.of(new EliminateCrossJoins())),
                 new PredicatePushDown(metadata, sqlParser),
+                new IterativeOptimizer(
+                        stats,
+                        new PickTableLayout(metadata, sqlParser, true).rules()),
```

Comment:
1. the rule runs once earlier 
2. I don't think we should pay attention to whether the rule was or was not run earlier.  Because someone could always put another instance of the rule earlier, and then setting this to false would be incorrect.  The point is that in this invocation of the rule we don't care whether a table layout has been set.  We always want to replace the layout.  This is different from the invocation after AddExchanges.  AddExchanges always chooses the "right" layout, so if there's a layout set, we don't want to overwrite it.